R/epidist_utils.R

```diff
@@ -0,0 +1,605 @@
+#' @param use_PMID A boolean logical determining whether the PMID is used in
+#' the citation
+#'
+#' @return A character string of the formatted short citation
+#' @export
+#'
+#' @examples
+#' create_epidist_citation(
+#'   author = "Smith_etal",
+#'   year = 2002,
+#'   DOI = "10.19832/j.1366-9516.2012.09147.x"
+#' )
+create_epidist_citation <- function(author = NA_character_,
+                                    year = NA_integer_,
+                                    PMID = NA_character_,
+                                    DOI = NA_character_,
+                                    use_PMID = FALSE) {
+  # check input
+  checkmate::assert_character(author)
+  checkmate::assert_number(year, na.ok = TRUE)
+  checkmate::assert_number(PMID, na.ok = TRUE)
+  checkmate::assert_character(DOI)
+  checkmate::assert_logical(use_PMID, len = 1)
+
+  if (is.na(author) || is.na(year) || is.na(DOI)) {
+    message(
+      "Citation cannot be created as either author, year or DOI is missing"
+    )
+    return("No citation available")
+  }
+
+  if (isTRUE(use_PMID) && is.na(PMID)) {
+    stop("use_PMID set to TRUE but PMID not provided")
+  }
+
+  # change author formatting if multiple authors or et al
+  author <- gsub(
+    pattern = "_",
+    replacement = " ",
+    x = author,
+    fixed = TRUE
+  )
+  author <- gsub(
+    pattern = "etal",
+    replacement = "et al.",
+    x = author,
+    fixed = TRUE
+  )
+
+  citation <- paste0(author, " (", year, ") ", "<", DOI, ">")
+
+  if (use_PMID) {
+    citation <- paste(citation, "PMID:", PMID)
+  }
+
+  citation
+}
+
+#' A helper function when creating an epidist object to create a method
+#' assessment list with sensible defaults, type checking and arguments to help
+#' remember which method assessments can be accepted in the list
+#'
+#' @param censorred A boolean logical whether the study used single or double
+#' interval censorring in the methods to infer the delay distribution
+#' @param right_truncated A boolean logical whether the study used right-
+#' truncation in the methods to infer the delay distribution
+#' @param phase_bias_adjusted A boolean logical whether the study adjusted for
+#' phase bias in the methods to infer the delay distribution
+#'
+#' @return A list of three elements
+#' @export
+#'
+#' @examples
+#' create_epidist_method_assessment(
+#'   censorred = FALSE,
+#'   right_truncated = FALSE,
+#'   phase_bias_adjusted = FALSE
+#' )
+create_epidist_method_assessment <- function(censorred = NA,
+                                             right_truncated = NA,
+                                             phase_bias_adjusted = NA) {
+  # check input
+  checkmate::assert_logical(censorred, len = 1)
+  checkmate::assert_logical(right_truncated, len = 1)
+  checkmate::assert_logical(phase_bias_adjusted, len = 1)
+
+  # return method assessment list
+  list(
+    censorred = censorred,
+    right_truncated = right_truncated,
+    phase_bias_adjusted = phase_bias_adjusted
+  )
+}
```

Comment:
1. Fix function title;
2. Typo in "censorred";
3. Value field should be expanded - typing this out gets repetitive and boring but better complete than vague.

Reply:
Updated documentation for `create_epidist_method_assessment()` in a1ba591. Fixed 'censor' typo throughout package.